Modified tile_items to export to absolute path

CakeJamble opened the pull request:
Changes were made in tile_items.py to allow the user to specify an absolute path to output the tiles and tile meta csv files. Instances of the base_path variable that was initialized as a global variable with pyprojroot have been removed now that the output path is specified by the user.

CakeJamble committed:
Modified tile_items.py to export to an absolute path. Exported miniset tiles for upload to Drive

cree321 commented:
~~Running, no issue.~~ *I know how branches work...* now.

cree321 commented:
parent_transform_from_file.py appears to work as intended.

nadia-eecs commented:
@campjake is mnt/d specific to the data directory mount in colab?

CakeJamble commented:
> @campjake is mnt/d specific to the data directory mount in colab?

We didn't notice that these empty directories were created locally as well! Thank you for checking that.
mnt is the name of the directory that holds the SSD and HDD's of the host machine when running Windows Subsystem for Linux. 

We output the tiles outside of the repository because the .gitignore file was changed to allow data to be uploaded for the augmentations team to experiment with the hotdog shaped tiles.

I will close this request since those empty directories are definitely a bug!

cree321 reviewed (changes requested):
### tile_items.py
A FileNotFound error is thrown on Windows for `cut_subset_tiles()`, line 163:
`temp_image.save(tile_f_name, "JPEG")`

`tile_f_name` is composed of duplicate paths since `self.parent_file_name` (line 226) is defined by `parent_path_input` (line 278). It seems [os.split() or pathlib.Path().name](https://stackoverflow.com/questions/8384737/extract-file-name-from-path-no-matter-what-the-os-path-format) could help to assign only the file name to parent_file_name, if this is in fact the issue.


Error:
> FileNotFoundError: [Errno 2] No such file or directory: 'C:/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/pre-processed\\20100905_000036_aia.lev1_euv_12s_4k/tiles/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/raw\\20100905_000036_aia.lev1_euv_12s_4k                    _tile_256_256.jpg'

Notes:
In `generate_tile_fpath_write()`, I verified `parent_path_input` = `/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/raw\20100905_000036_aia.lev1_euv_12s_4k.jpg`

Initial paths (lmk if I just made a mistake):
`PARENT_IMAGE_INPUT_PATH = 'C:/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data'`
`TILE_OUTPUT_PATH = 'C:/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/pre-processed/latest4096'`

Good work! It just may need a tiny bit more attention.

CakeJamble commented:
> ### tile_items.py
> 
> A FileNotFound error is thrown on Windows for `cut_subset_tiles()`, line 163: `temp_image.save(tile_f_name, "JPEG")`
> 
> `tile_f_name` is composed of duplicate paths since `self.parent_file_name` (line 226) is defined by `parent_path_input` (line 278). It seems [os.split() or pathlib.Path().name](https://stackoverflow.com/questions/8384737/extract-file-name-from-path-no-matter-what-the-os-path-format) could help to assign only the file name to parent_file_name, if this is in fact the issue.
> 
> Error:
> 
> > FileNotFoundError: [Errno 2] No such file or directory: 'C:/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/pre-processed\20100905_000036_aia.lev1_euv_12s_4k/tiles/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/raw\20100905_000036_aia.lev1_euv_12s_4k                    _tile_256_256.jpg'
> 
> Notes: In `generate_tile_fpath_write()`, I verified `parent_path_input` = `/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/raw\20100905_000036_aia.lev1_euv_12s_4k.jpg`
> 
> Initial paths (lmk if I just made a mistake): `PARENT_IMAGE_INPUT_PATH = 'C:/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data'` `TILE_OUTPUT_PATH = 'C:/Users/trazm/Documents/hits-sdo-packager-batch-tiling/hits-sdo-packager/data/pre-processed/latest4096'`
> 
> Good work! It just may need a tiny bit more attention.

I added a quick fix that will normalize the path following the line you mentioned. I suspect there will still be issues with that in other functions. If you are testing this again and see this issue elsewhere, can you open a new issue?
